tests/test_model.py

```diff
@@ -28,79 +28,95 @@
 
 
 class ModelTestCase(unittest.IsolatedAsyncioTestCase):
-    """
-    tests the Mock T2SA class. This is a simple stand-in for
-    the T2SA application that we can connect to, and which has
-    canned responses to some simple alignment system commands.
-    """
+    log: logging.Logger
 
     @classmethod
-    def setUpClass(cls):
-        cls.log = logging.getLogger()
-        cls.log.addHandler(logging.StreamHandler())
-        cls.log.setLevel(logging.INFO)
+    def setUpClass(cls) -> None:
+        cls.log = logging.getLogger(__name__)
 
-    async def asyncSetUp(self):
+    async def asyncSetUp(self) -> None:
         self.mock_t2sa = MTAlignment.MockT2SA(port=0, log=self.log)
+        # Set warmup time to 1.0 second to speed up testing.
+        self.mock_t2sa.laser_warmup_time = 1.0
         await asyncio.wait_for(self.mock_t2sa.start_task, 5)
 
-    async def asyncTearDown(self):
+    async def asyncTearDown(self) -> None:
         if self.mock_t2sa:
             await asyncio.wait_for(self.mock_t2sa.close(), 5)
 
-    async def test_connect(self):
-        """
-        Tests we can connect to the mock T2SA server
-        """
+    async def test_connect(self) -> None:
+        """Tests we can connect to the mock T2SA server."""
         self.model = MTAlignment.AlignmentModel(
             host=LOCAL_HOST,
             port=self.mock_t2sa.port,
             read_timeout=30,
-            simulation_mode=1,
+            t2sa_simulation_mode=1,
             log=self.log,
         )
         await self.model.connect()
         assert self.model.connected
         await self.model.disconnect()
         assert not self.model.connected
 
-    async def test_laser_status(self):
-        """
-        Tests mock T2SA reports laser status as "on"
-        """
+    async def test_laser_status(self) -> None:
+        """Tests mock T2SA reports laser status as "on"."""
         self.model = MTAlignment.AlignmentModel(
             host=LOCAL_HOST,
             port=self.mock_t2sa.port,
             read_timeout=30,
-            simulation_mode=1,
+            t2sa_simulation_mode=1,
             log=self.log,
         )
         await self.model.connect()
+        self.log.debug("Query initial laser status. should be LOFF")
+        response = await self.model.send_command("?LSTA")
+        assert response == "LOFF"
+
+        self.log.debug("Power-up laser.")
+        response = await self.model.send_command("!LST:1")
+
+        self.log.debug("Query laser status while warming up, should be WARM.")
+        response = await self.model.send_command("?LSTA")
+        assert "WARM" in response
+
+        self.log.debug("Waiting for laser to finish warming up.")
+        await self.mock_t2sa.laser_warmup_task
```

Comment:
I suggest using `await asyncio.wait_for` whenever there is a chance the thing you are waiting for might never finish. It is messier, but it makes test results much easier to understand if something goes wrong: instead of hanging forever (and having no idea where it hung), the test times out with a traceback.